src/tally.F90

```diff
@@ -2740,6 +2756,712 @@ subroutine score_surface_current(p)
+    type(Particle),    intent(in)    :: p
+    type(TallyObject), intent(in)    :: t
+    integer,           intent(in)    :: i_nuclide
+    real(8),           intent(in)    :: atom_density   ! atom/b-cm
+    integer,           intent(in)    :: score_bin
+    real(8),           intent(inout) :: score
+
+    integer :: l
+    logical :: scoring_diff_nuclide
+    real(8) :: flux_deriv
+    real(8) :: dsigT, dsigA, dsigF, cum_dsig
+
+    if (score == ZERO) return
+
+    flux_deriv = p % flux_derivs(t % deriv)
+
+    associate(deriv => tally_derivs(t % deriv))
+      select case (tally_derivs(t % deriv) % variable)
+
+      case (DIFF_DENSITY)
+        select case (t % estimator)
+
+        case (ESTIMATOR_ANALOG)
+          if (materials(p % material) % id == deriv % diff_material) then
+            score = score * (flux_deriv + ONE &
+                 / materials(p % material) % density_gpcc)
+          else
+            score = score * flux_deriv
+          end if
+
+        case (ESTIMATOR_COLLISION)
+
+          select case (score_bin)
+
+          case (SCORE_FLUX)
+            score = score * flux_deriv
+
+          case (SCORE_TOTAL)
+            if (materials(p % material) % id == deriv % diff_material &
+                 .and. material_xs % total /= ZERO) then
+              score = score * (flux_deriv + ONE &
+                   / materials(p % material) % density_gpcc)
+            else
+              score = score * flux_deriv
+            end if
+
+          case (SCORE_SCATTER)
+            if (materials(p % material) % id == deriv % diff_material &
+                 .and. material_xs % total - material_xs % absorption /= ZERO) &
+                 then
+              score = score * (flux_deriv + ONE &
+                   / materials(p % material) % density_gpcc)
+            else
+              score = score * flux_deriv
+            end if
+
+          case (SCORE_ABSORPTION)
+            if (materials(p % material) % id == deriv % diff_material &
+                 .and. material_xs % absorption /= ZERO) then
+              score = score * (flux_deriv + ONE &
+                   / materials(p % material) % density_gpcc)
+            else
+              score = score * flux_deriv
+            end if
+
+          case (SCORE_FISSION)
+            if (materials(p % material) % id == deriv % diff_material &
+                 .and. material_xs % fission /= ZERO) then
+              score = score * (flux_deriv + ONE &
+                   / materials(p % material) % density_gpcc)
+            else
+              score = score * flux_deriv
+            end if
+
+          case (SCORE_NU_FISSION)
+            if (materials(p % material) % id == deriv % diff_material &
+                 .and. material_xs % nu_fission /= ZERO) then
+              score = score * (flux_deriv + ONE &
+                   / materials(p % material) % density_gpcc)
+            else
+              score = score * flux_deriv
+            end if
+
```

Comment:
I completely agree that it can be laborious to verity the correctness of those tallies, especially `DELAYED_NU_FISSION`! I don't have a use case for any of the four I mentioned, but I would be interested in the surface current score. That one is implemented somewhat differently than the standard tallies though. Are surface current tally derivatives supported?